packages/forklift-console-plugin/src/modules/Mappings/data.ts

```diff
@@ -0,0 +1,202 @@
+        destination as INameNamespaceRef,
+        source,
+      ]),
+    ),
+  ];
+};
+
+interface OwnerRef {
+  name?: string;
+  gvk?: K8sGroupVersionKind;
+}
+
+const resolveOwnerRef = ([first, second]: OwnerReference[] = []): OwnerRef => {
+  // expect only one owner - the plan
+  if (!first || first.kind != ResourceKind.Plan || second) {
+    return {};
+  }
+
+  return { name: first.name, gvk: groupVersionKindForObj(first) };
+};
+
+const mergeData = (
+  mappings: NetworkMapResource[],
+  providers: ProviderResource[],
+): FlatNetworkMapping[] => {
+  return mappings
+    .map(
+      (
+        mapping,
+      ): [
+        NetworkMapResource,
+        NetworkMapResource,
+        K8sGroupVersionKind,
+        ProviderRef,
+        ProviderRef,
+        [Network, IdOrNameRef[]][],
+        OwnerRef,
+      ] => [
+        mapping, // to extract props
+        mapping, // to pass as object blob
+        groupVersionKindForObj(mapping),
+        resolveProviderRef(mapping.spec.provider.source, providers),
+        resolveProviderRef(mapping.spec.provider.destination, providers),
+        // future improvement: resolve GVK for local networks:
+        // 1. target provider needs to be local
+        // 2. fetch available networks - requires 'kind' of CRDs created by Multus CNI
+        groupByTarget(mapping),
+        resolveOwnerRef(mapping.metadata.ownerReferences),
+      ],
+    )
+    .map(
+      ([
+        {
+          metadata: { name, namespace, annotations = [] },
+        },
+        mapping,
+        gvk,
+        sourceProvider,
+        targetProvider,
+        groupedNetworks,
+        owner,
+      ]): FlatNetworkMapping => ({
+        name,
+        namespace,
+        template: annotations?.[C.SHARED_MAPPING_ANNOTATION] !== 'false',
+        gvk,
+        owner: owner.name,
+        ownerGvk: owner.gvk,
+        source: sourceProvider.name,
+        sourceGvk: sourceProvider.gvk,
+        sourceResolved: sourceProvider.resolved,
+        sourceReady: sourceProvider.ready,
+        target: targetProvider.name,
+        targetGvk: targetProvider.gvk,
+        targetResolved: targetProvider.resolved,
+        targetReady: targetProvider.ready,
+        from: groupedNetworks,
+        to: groupedNetworks.map(([to]) => to),
+        object: mapping,
+      }),
+    )
+    .filter((it) => it.template);
+};
+
+export interface LocalNetworkResource {
+  gvk: K8sGroupVersionKind;
+  name: string;
+  namespace: string;
+  type: 'multus';
+}
+
+export interface RemoteNetworkResource {
+  name: string;
+  namespace: string;
+  type: 'multus';
+}
+
+export interface PodNetwork {
+  name: '';
+  type: 'pod';
+}
+
+export type Network = LocalNetworkResource | RemoteNetworkResource | PodNetwork;
+
+export const useFlaNetworkMappings = ({
+  namespace,
+  name = undefined,
+  groupVersionKind: { group, version },
+}): [FlatNetworkMapping[], boolean, boolean] => {
+  const [providers] = useProviders({ namespace }, { group, version });
```

Comment:
you should ONLY get the net mappings,
if you need to load something else, then you may show wrong data ?
can you check if we can you show a nice mapping row without extra fetching ?

mapping has:

> apiVersion: forklift.konveyor.io/v1beta1
> kind: NetworkMap
> metadata:
>   name: <network_map>
>   namespace: konveyor-forklift
> spec:
>   map:
>     - destination:
>         name: <pod>
>         namespace: konveyor-forklift
>         type: pod (1)
>       source: (2)
>         id: <source_network_id> (3)
>         name: <source_network_name>
>     - destination:
>         name: <network_attachment_definition> (4)
>         namespace: <network_attachment_definition_namespace> (5)
>         type: multus
>       source:
>         id: <source_network_id>
>         name: <source_network_name>
>   provider:
>     source:
>       name: <source_provider>
>       namespace: konveyor-forklift
>     destination:
>       name: <destination_cluster>
>       namespace: konveyor-forklift
> EOF

so we can show:
name, namespace, source and destination, + some nice table of from -> to networks

Reply:
showing nice status that take into account state of nets and providers is nice to have, we can add it later when we decide what is the best way to do it

Reply:
that sounds strange?
we have links to providers, we know there GVK , what am i missing?

Reply:
The source/target pseudo-reference in NetworkMap resource is not sufficient to build a resource link. 
It contains only name and namespace. AFAIK the resource link requires full GVK (for CRDs).

Reply:
Actually for proper linking we should use `ownerRefs` and/or UIDs. Without that UI-based references are not reliable i.e. try:
1. create a source provider 'test' pointing to cluster B
2. create migration plan from B -> sth
3. delete provider 'test'
4. create provider 'test' pointing to cluster C

If  source is using some popular network/VM names like `test` then it could actually still work:)

Reply:
if you know it's a `provider` and you know `name` and `namespace`, that is all you need, if it's not, lets not have links for now, and fix it in a nice way latter

Reply:
> 
> 
> Actually for proper linking we should use ownerRefs and/or UIDs. Without that UI-based references are not reliable i.e. try:
> 
>     create a source provider 'test' pointing to cluster B
>     create migration plan from B -> sth
>     delete provider 'test'
>     create provider 'test' pointing to cluster C
> 
> If source is using some popular network/VM names like test then it could actually still work:)
> 

AFAIU in this case you can not use owner ref, because the plan will point to the new provider (that has a different id) the plan will use name+namespace to find the provider not the id

Reply:
1. dropped validating network references as this requires a new query on the backend (a  2.4.1 candidate?)
2. keep validating provider refs as we can get the changes effectively using k8s watch API